firebase-components/src/main/java/com/google/firebase/components/ComponentRuntime.java

```diff
@@ -107,6 +108,8 @@ private void discoverComponents(List<Component<?>> componentsToAdd) {
     // instead of executing such code in the synchronized block below, we store it in a list and
     // execute right after the synchronized section.
     List<Runnable> runAfterDiscovery = new ArrayList<>();
+    List<Component<?>> componentsAdding = new ArrayList<>();
```

Comment:
There's one small behavioural change which out of precaution I think we should restore. The current code works as follows

1. Call `discoverComponents` with the list of `componentsToAdd`
2. During execution, we process pending providers, **and modify** `componentsToAdd`
3. We ensure the list is ok using the `CycleDetector`
4. Do what needs to be done with the components

This change no longer updates `componentsToAdd` to include what is actually going to be processed.


. In the current version, we get as parameter `componentsToAdd` which is the list of things we have to add